frontend/src/pages/Clusters/Logs.js

```diff
@@ -12,50 +12,91 @@ import React from 'react';
+
+  const pending = useState(['app', 'clusters', 'logs', 'pending']);
+
+  const refresh = () => {
+    setState(['app', 'clusters', 'logs', 'pending'], true);
+    const clusterName = getState(['app', 'clusters', 'selected']);
+    const logStreamName = getState(['app', 'clusters', 'selectedLogStreamName']);
+    if(clusterName && logStreamName)
+    {
+      GetClusterLogEvents(clusterName, logStreamName, () => clearState(['app', 'clusters', 'logs', 'pending']), () => clearState(['app', 'clusters', 'logs', 'pending']));
+    }
+  }
+
+  return <div><div style={{marginBottom: "10px", display: "flex", direction: "row", gap: "16px", alignItems: "center"}}><div>{selectedLogStreamName}</div><Button loading={pending} onClick={refresh}>Refresh</Button></div>
+    <div style={{borderTop: "1px solid #AAA", fontSize: "10pt", overflow: "auto", whiteSpace: "nowrap"}}>
+      {events.events.map((event, i) => <div key={event.timestamp + i.toString()} title={event.timestamp}>{event.message}</div>)}
     </div>
-  );
+  </div>
 }
 
-export default function ClusterLogs() {
+function StreamList({instanceId}) {
+  const logStreamIndex = useState(['app', 'clusters', 'logs', 'index']);
+  const logStreams = logStreamIndex[instanceId].streams;
+  const ip = logStreamIndex[instanceId].ip;
+  const fnames = Object.keys(logStreams).sort()
+  const selectedLogStreamName = useState(['app', 'clusters', 'selectedLogStreamName']);
 
-  const [ split, setSplit ] = React.useState(80);
-  const [ isSelected , setSelected ] = React.useState(false);
+  const select = (logStream) => {
+    const logStreamName = logStream.logStreamName;
+    const selected = getState(['app', 'clusters', 'selected']);
+    setState(['app', 'clusters', 'selectedLogStreamName'], logStreamName);
+    GetClusterLogEvents(selected, logStreamName);
+  }
 
-  const selected = useState(['app', 'clusters', 'selected']);
-  const streams = useState(['clusters', 'index', selected, 'logstreams']);
-  const selectedLogStreamName = useState(['app', 'clusters', 'selectedLogStreamName']);
-  const logEvents = useState(['clusters', 'index', selected, 'logEventIndex', selectedLogStreamName]);
+  return <div title={instanceId}>
+    <ExpandableSection header={ip}>
+      {fnames.map((fname) => <div onClick={() => select(logStreams[fname])} style={{marginLeft: '10px', cursor: 'pointer', fontWeight: selectedLogStreamName === logStreams[fname].logStreamName ? 'bold' : 'normal'}}>{fname}</div>)}
+    </ExpandableSection>
+  </div>
+}
 
+function LogStreamList() {
+  const logStreamIndex = useState(['app', 'clusters', 'logs', 'index']);
+  const selected = useState(['app', 'clusters', 'selected']);
   const clusterPath = ['clusters', 'index', selected];
   const headNode = useState([...clusterPath, 'headNode']);
   const instanceId = (headNode && headNode.instanceId) || '';
+  return <div style={{width: "150px"}}>
+    <div><b>HeadNode</b></div>
```

Comment:
Can we use the `awsui` ColumnLayout here?

Reply:
Unfortunately we don't have the detailed control over the width of the columns using that component.